Rakefile

```diff
@@ -1,42 +1,76 @@
-%w{ bundler/setup rubygems fileutils uri net/https tmpdir digest/md5 ./doc/markit }.each do |file|
+%w{ bundler/setup yaml uri net/https tmpdir digest/md5 ./doc/markit }.each do |file|
   require file
 end
 
 # Immediately sync all stdout so that it's immediately visible, e.g. on appveyor
 $stdout.sync = true
 $stderr.sync = true
 
-VERSION = '3.1-SNAPSHOT'
-BASE_DIR = File.expand_path('.', File.dirname(__FILE__))
-TARGET_DIR  = "#{BASE_DIR}/target"
-BUILD_DIR   = "#{BASE_DIR}/target/build"
-CACHE_DIR   = "#{BASE_DIR}/target/cache"
-ZIP_EXE = 'C:\Program Files\7-Zip\7z.exe'
+module EnvironmentOptions
+  BASE_DIR = File.expand_path('.', File.dirname(__FILE__))
+  VERSION = '3.1-SNAPSHOT'
+  def version
+    VERSION
+  end
+  def base_dir
+    return BASE_DIR
+  end
+  def target_dir
+    ddir=ENV["TARGET_DIR"]
+    ddir||=File.join(base_dir,"target")
+    File.expand_path(ddir)
+  end
+  def build_dir
+    ddir=ENV["BUILD_DIR"]
+    ddir||=File.join(base_dir,"build")
+    File.expand_path(ddir)
+  end
+  def cache_dir
+    ddir=ENV["CACHE_DIR"]
+    ddir||=File.join(base_dir,"cache")
+    File.expand_path(ddir)
+  end
+  def zip_exe
+    zip=ENV["ZIP_EXE"]
+    zip||='C:\Program Files\7-Zip\7z.exe'
+    return File.expand_path(zip)
+  end
+end
 
+include EnvironmentOptions
 
 desc 'cleans the build output directory'
 task :clean do
   purge_atom_plugins_with_insanely_long_path
-  FileUtils.rm_rf BUILD_DIR, secure: true
+  FileUtils.rm_rf build_dir, secure: true
 end
 
 desc 'wipes all output and cache directories'
 task :wipe do
   purge_atom_plugins_with_insanely_long_path
-  FileUtils.rm_rf TARGET_DIR, secure: true
+  FileUtils.rm_rf target_dir, secure: true
 end
 
 desc 'downloads required resources and builds the devpack binary'
 task :build => :clean do
+  tools_config=YAML.load(File.read("#{base_dir}/config/tools.yaml"))
   recreate_dirs
-  download_tools
-  move_chefdk
-  fix_chefdk
+  download_tools(tools_config)
+  if tools_config.keys.include?("chefdk")
```

Comment:
wonder if we could make this a bit more generic and pluggable. not necessarily for bills-kitchen here, but if we want to use that as a minimal template for future devpacks